scripts/task/KvToRdsMigration.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+
+namespace oat\taoOutcomeUi\scripts\task;
+
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\taoDelivery\model\execution\ServiceProxy;
+use oat\taoOutcomeRds\model\RdsResultStorage;
+use oat\taoResultServer\models\classes\implementation\ResultServerService;
+
+class KvToRdsMigration extends ScriptAction
+{
+    use OntologyAwareTrait;
+
+    /** @var RdsResultStorage $kvStorageService */
+    protected $rdsStorage;
+
+    /** @var \taoResultServer_models_classes_ReadableResultStorage $kvStorageService */
+    protected $kvStorage;
+
+    /** @var array  */
+    protected $deliveryExecutions = [];
+
+    /** @var \common_report_Report */
+    protected $report;
+
+    /**
+     * Provide a "force" flag to apply the migration
+     *
+     * @return array
+     */
+    protected function provideOptions()
+    {
+        return [
+            'force' => [
+                'prefix' => 'f',
+                'longPrefix' => 'force',
+                'required' => false,
+                'default' => true,
+                'flag' => true,
+                'description' => 'Apply the migration to Result storage.',
+            ]
+        ];
+    }
+
+    /**
+     * Run the command.
+     *
+     * Get current result storage as key value storage
+     * Retrieve result $callIds and migrate deliveryExecution
+     * Migrate the result storage to rdsStorage
+     *
+     * @return \common_report_Report The report of the migration
+     * @throws \common_Exception
+     */
+    protected function run()
+    {
+        $this->kvStorage = $this->getCurrentKvResultStorage();
+
+        $this->rdsStorage = $this->getServiceLocator()->get(RdsResultStorage::SERVICE_ID);
+
+        $this->report = \common_report_Report::createInfo('Starting migration...');
+
+        foreach ($this->kvStorage->getAllCallIds() as $callId) {
+            foreach ($this->kvStorage->getVariables($callId) as $variables) {
+                $this->migrateDeliveryExecution($callId, $variables);
+            }
+        }
+
+        $this->setCurrentResultStorageToRdsStorage();
```

Comment:
It would be good to add a report entry for this as well, so that people know the Storage type will be updated as well, eg. `$this->report->add(\common_report_Report::createSuccess('Storage type successfully updated to RDS Storage'))`